### PR TITLE
Readme updates for other future contributors + Basic Books and Shelves Generator

### PR DESCRIPTION
**Pull request by NoahTheCorgi**
1. Readme files in dynamic and generators directories were updated to contain more specific instructions on how to contribute for future contributors

2. Readme files in dynamic and generators directories were updated to contain information about respecting copyright and course materials as well as receiving permission from owners accordingly.

3. Basic Books and Shelves Generator Implemented under booksinshelves directory within generators

4. Updated the test_main.py file so that the randomly generated call() method does not conflict with a fixed test output.

### CODE REVIEW

**Comment by NoahTheCorgi** [comment]
The code now randomly generates three possible types of "books in shelves" questions.

The generate_question() in main.py now randomly creates the following three types of books in shelves questions.

1. How many ways to organize $a$ books in $b$ number of book shelves?

2. How many ways to organize $a$ books in $n$ out of $b$ bookshelves? (i.e. exactly $n$ out of $b$ shelves must be used.)

3. How many ways to organize $a$ books in $b$ number of bookshelves? ALL shelves must be used (i.e. there cannot be empty bookshelves.)

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,4 @@ stories
 __pycache__
 
 .DS_Store
+dynamic/generators/booksinshelves/booksinshelves2.py
```
Comment (AngelOnFira) [inline .gitignore:9]: Why was this added to the gitignore?

Comment (NoahTheCorgi) [reply .gitignore]: Thank you for pointing this out! I will make that the habit. I just updated it accordingly.

**Comment by NoahTheCorgi** [comment]
Previously passed versions of the test_out.py or test_output.py all fail the checks, which suggests that possibly, there is a bug in the checks themselves. I will look more into my code also. It would be great to know if there recently has been some changes in the checks performed for the test-dynamic.

**Review by MathyouMB — APPROVED** [review]

**Comment by NoahTheCorgi** [comment]
Thank you!
